src/core/components.py

```diff
@@ -341,6 +341,48 @@ def get_base_component_dict(self):
             "destination_path": self.destination_path,
         }
 
+class Selection(Component):
+    __slots__ = [
+        "style",
+        "preconditions",
+        "destination_path",
+        "options_key",
+        "options_values",
+    ]
+
+    label = Translatable()
+
+    def get_options(self, options_key, options_values):
+        """Only key or values should be specified, this function validates this condition"""
+        if options_key is not None and options_values is not None:
+            raise InvalidArguments(
+                "'options_key' and 'options_values' attribute cannot be used together"
+            )
+
+        if options_key is None and options_values is None:
+            raise InvalidArguments("Either 'options_key' or 'options_values' attribute must be used")
+        
+        if options_values and type(options_values) is not dict:
+            raise InvalidArguments("If 'options_values' is supplied, it must be a dictionary")
+        
+        return options_key, options_values
+
+    def __init__(self, style, label, value=None, destination_path=None, options_key=None, options_values=None, **kwargs):
+        super().__init__(**kwargs)
+        self.style = style
+        self.label = label
+        self.options_key, self.options_values = self.get_options(options_key, options_values)
```

Comment:
validate_options

Reply:
it gets the options as well, validate_and_get_options as a compromise?

Reply:
to be honest I think get_options() was better, it's expected that a method will validate it's own parameters and raise errors if something is wrong

Reply:
oh saying that, validate_and_get_options isn't returning anything?

Reply:
reverted the change back to single get which validates

Reply:
Wait why was it called `validate_and_get_options` and not just `validate_options`